src/runtime/math.py

```diff
@@ -215,25 +227,68 @@ def fabs(x: f64) -> f64:
         return -x
     return x
 
+@overload
+def fabs(x: i64) -> i64:
+    """
+    Return the absolute value of `x`.
+    """
+    if x < 0:
+        return -x
+    return x
 
-def ldexp(x: f64, i: i32) -> f64:
-    return x * (2**i)
+@overload
+def fabs(x: i32) -> i32:
+    """
+    Return the absolute value of `x`.
+    """
+    if x < 0:
+        return -x
+    return x
 
+# pow
+# supported data types: i32, i64, f32, f64
+@overload
+def pow(x: f64, y: f64) -> f64:
+    """
+    Return `x` raised to the power `y`.
+    """
+    return x**y
 
-def exp(x: f64) -> f64:
+
+@overload
+def pow(x: i64, y: i64) -> i64:
     """
-    Return `e` raised to the power `x`.
+    Return `x` raised to the power `y`.
     """
-    return e**x
+    return x**y
 
+@overload
+def pow(x: f32, y: f32) -> f64:
+    """
+    Return `x` raised to the power `y`.
+    """
+    return x**y
 
-def pow(x: f64, y: f64) -> f64:
+@overload
+def pow(x: i32, y: i32) -> i64:
     """
     Return `x` raised to the power `y`.
     """
     return x**y
```

Comment:
```suggestion
    assert y >= 0
    return x**y
```

Use only if `ValueError` fails to work.